components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientResourceImplTest.java

```diff
@@ -0,0 +1,358 @@
+
+    private String id = "00000001";
+
+
+    @Before
+    public void setUp() throws ComponentLookupException, URISyntaxException {
+        MockitoAnnotations.initMocks(this);
+        Execution execution = mock(Execution.class);
+        ExecutionContext executionContext = mock(ExecutionContext.class);
+
+        ComponentManager componentManager = this.mocker.getInstance(ComponentManager.class, "context");
+        when(componentManager.getInstance(Execution.class)).thenReturn(execution);
+        doReturn(executionContext).when(execution).getContext();
+        doReturn(mock(XWikiContext.class)).when(executionContext).getProperty("xwikicontext");
+
+        this.patientResource = (DefaultPatientResourceImpl)this.mocker.getComponentUnderTest();
+        this.logger = this.mocker.getMockedLogger();
+        this.repository = this.mocker.getInstance(PatientRepository.class);
+        this.access = this.mocker.getInstance(AuthorizationManager.class);
+        this.users = this.mocker.getInstance(UserManager.class);
+
+
+        doReturn(this.currentUser).when(this.users).getCurrentUser();
+        doReturn(null).when(this.currentUser).getProfileDocument();
+
+        doReturn(this.patient).when(this.repository).getPatientById(this.id);
+        doReturn(null).when(this.patient).getDocument();
+
+        this.uri = new URI(this.uriString);
+        doReturn(this.uri).when(this.uriInfo).getRequestUri();
+
+        ReflectionUtils.setFieldValue(this.patientResource, "uriInfo", this.uriInfo);
+    }
+
+    //----------------------------Get Patient Tests----------------------------
+
+    @Test
+    public void getPatientCantFindPatient() throws XWikiRestException {
+        doReturn(null).when(this.repository).getPatientById(anyString());
+        Response response = this.patientResource.getPatient(this.id);
+
+        verify(this.logger).debug("No such patient record: [{}]", this.id);
+        Assert.assertEquals(Status.NOT_FOUND.getStatusCode(), response.getStatus());
+    }
+
+    @Test
+    public void getPatientUserDoesNotHaveAccess() throws XWikiRestException {
+        doReturn(false).when(this.access).hasAccess(Right.VIEW, null, null);
+
+        Response response = this.patientResource.getPatient(this.id);
+
+        verify(this.logger).debug("View access denied to user [{}] on patient record [{}]", this.currentUser, this.id);
+        Assert.assertEquals(Status.FORBIDDEN.getStatusCode(), response.getStatus());
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void getPatientAddsLinksToPatientJSON() throws XWikiRestException {
+        doReturn(true).when(this.access).hasAccess(Right.VIEW, null, null);
+        doReturn(new JSONObject()).when(this.patient).toJSON();
+
+        Response response = this.patientResource.getPatient(this.id);
+
+        Assert.assertTrue(response.getEntity() instanceof JSONObject);
+
+        Map<String, Map<String, String>> json = (Map)response.getEntity();
```

Comment:
The `org.hamcrest.Matchers.hasEntry` and `hasValue` expect a Map. JSONObject implements the `java.uitl.Map` interface, but there will be an error unless the JSONObject is first cast to a Map